src/Symfony/Component/Asset/VersionStrategy/JsonManifestVersionStrategy.php

```diff
@@ -46,23 +51,26 @@ public function getVersion(string $path)
     }
 
     public function applyVersion(string $path)
-    {
-        return $this->getManifestPath($path) ?: $path;
-    }
-
-    private function getManifestPath(string $path): ?string
     {
         if (null === $this->manifestData) {
             if (!is_file($this->manifestPath)) {
-                throw new \RuntimeException(sprintf('Asset manifest file "%s" does not exist.', $this->manifestPath));
+                throw new RuntimeException(sprintf('Asset manifest file "%s" does not exist.', $this->manifestPath));
             }
 
             $this->manifestData = json_decode(file_get_contents($this->manifestPath), true);
             if (0 < json_last_error()) {
-                throw new \RuntimeException(sprintf('Error parsing JSON from asset manifest file "%s": ', $this->manifestPath).json_last_error_msg());
+                throw new RuntimeException(sprintf('Error parsing JSON from asset manifest file "%s": ', $this->manifestPath).json_last_error_msg());
             }
         }
 
-        return isset($this->manifestData[$path]) ? $this->manifestData[$path] : null;
+        if (isset($this->manifestData[$path])) {
+            return $this->manifestData[$path];
+        }
+
+        if ($this->strict) {
+            throw new RuntimeException(sprintf('Asset "%s" not found in manifest "%s".', $path, $this->manifestPath));
+        }
+
+        return $path;
```

Comment:
We previously returned null in this situation, right?

The remote version DOES return path, but this one returned null (I can’t remember why that difference exists at the moment, but for BC it should remain how it was)

Reply:
I merged `getManifestPath` into `applyVersion`. The later returns a string.